miniapps/nurbs/nurbs_ex24.cpp

```diff
@@ -0,0 +1,485 @@
+   //    corresponding to the trial fespace.
+   GridFunction gftest(&test_fes);
+   GridFunction gftrial(&trial_fes);
+   GridFunction x(&test_fes);
+   FunctionCoefficient p_coef(p_exact);
+   VectorFunctionCoefficient gradp_coef(sdim, gradp_exact);
+   VectorFunctionCoefficient v_coef(sdim, v_exact);
+   VectorFunctionCoefficient curlv_coef(sdim, curlv_exact);
+   FunctionCoefficient divgradp_coef(div_gradp_exact);
+
+   if (prob == 0)
+   {
+      gftrial.ProjectCoefficient(p_coef);
+   }
+   else if (prob == 1)
+   {
+      gftrial.ProjectCoefficient(v_coef);
+   }
+   else
+   {
+      gftrial.ProjectCoefficient(gradp_coef);
+   }
+
+   gftrial.SetTrueVector();
+   gftrial.SetFromTrueVector();
+
+   // 7. Set up the bilinear forms for L2 projection.
+   ConstantCoefficient one(1.0);
+   BilinearForm a(&test_fes);
+   MixedBilinearForm a_mixed(&trial_fes, &test_fes);
+   if (pa)
+   {
+      a.SetAssemblyLevel(AssemblyLevel::PARTIAL);
+      a_mixed.SetAssemblyLevel(AssemblyLevel::PARTIAL);
+   }
+
+   if (prob == 0)
+   {
+      a.AddDomainIntegrator(new VectorFEMassIntegrator(one));
+      a_mixed.AddDomainIntegrator(new MixedVectorGradientIntegrator(one));
+   }
+   else if (prob == 1)
+   {
+      a.AddDomainIntegrator(new VectorFEMassIntegrator(one));
+      a_mixed.AddDomainIntegrator(new MixedVectorCurlIntegrator(one));
+   }
+   else
+   {
+      a.AddDomainIntegrator(new MassIntegrator(one));
+      a_mixed.AddDomainIntegrator(new VectorFEDivergenceIntegrator(one));
+   }
+
+   // 8. Assemble the bilinear form and the corresponding linear system,
+   //    applying any necessary transformations such as: eliminating boundary
+   //    conditions, applying conforming constraints for non-conforming AMR,
+   //    static condensation, etc.
+   if (static_cond) { a.EnableStaticCondensation(); }
+
+   a.Assemble();
+   if (!pa) { a.Finalize(); }
+
+   a_mixed.Assemble();
+   if (!pa) { a_mixed.Finalize(); }
+
+   if (pa)
+   {
+      a_mixed.Mult(gftrial, x);
+   }
+   else
+   {
+      SparseMatrix& mixed = a_mixed.SpMat();
+      mixed.Mult(gftrial, x);
+   }
+
+   // 9. Define and apply a PCG solver for Ax = b with Jacobi preconditioner.
+   {
+      GridFunction rhs(&test_fes);
+      rhs = x;
+      x = 0.0;
+
+      CGSolver cg;
+      cg.SetRelTol(1e-12);
+      cg.SetMaxIter(1000);
+      cg.SetPrintLevel(1);
+      if (pa)
+      {
+         Array<int> ess_tdof_list; // empty
+         OperatorJacobiSmoother Jacobi(a, ess_tdof_list);
+
+         cg.SetOperator(a);
+         cg.SetPreconditioner(Jacobi);
+         cg.Mult(rhs, x);
+      }
+      else
+      {
+         SparseMatrix& Amat = a.SpMat();
+         DSmoother Jacobi(Amat);
+
+         cg.SetOperator(Amat);
+         cg.SetPreconditioner(Jacobi);
+         cg.Mult(rhs, x);
+      }
+   }
+
+   // 10. Compute the same field by applying a DiscreteInterpolator.
+   /* DicreteInterpolants are not implemented yet (NIY) for NURBS
```

Comment:
Should this commented section be removed?

Reply:
Yes. I was thinking to perhaps make the functionality work for NURBS. But I think the cost/benefit makes it not worth it. So might as well delete this stuff.